pkg/logs/agent.go

```diff
@@ -34,12 +36,15 @@ type Agent struct {
 	destinationsCtx  *sender.DestinationsContext
 	pipelineProvider pipeline.Provider
 	inputs           []restart.Restartable
+	health           *health.Handle
 }
 
 // NewAgent returns a new Agent
 func NewAgent(sources *config.LogSources, services *service.Services, endpoints *config.Endpoints) *Agent {
+	h := health.Register("logs-agent")
+
 	// setup the auditor
-	auditor := auditor.New(config.LogsAgent.GetString("logs_config.run_path"))
+	auditor := auditor.New(config.LogsAgent.GetString("logs_config.run_path"), h)
```

Comment:
Why are we adding the health check to the auditor? What are the alternatives?

Reply:
I chose the auditor because it's the end of the pipeline. Arguably it could also be plugged to the main destination (when a connection is open?), do you think it's a better idea?

Reply:
+1 for the auditor 
I know how the health works but I think it's worth adding a comment here to explain what's the inner logic and why it makes more sense to add it to the auditor.

Reply:
added a comment.